### ISSUE TITLE
[Bug] Gen 2 hamburger menu closes automatically when search input field entered at certain mobile resolutions

### ISSUE THREAD

**Issue by jaredh-um** [unimelb/pattern-lib#1564]
### Expected behaviour

Users on mobile should be able to open the hamburger menu, tap into the search field, enter a query and then perform a search, with the menu remaining visible until the search results page is loaded.

![UoM Mobile search.jpg](https://images.zenhubusercontent.com/5acae0cd4b5806bc2bce3338/3f274219-a9e8-47af-b0f8-1f0814bd2cb0)

### Current behaviour

When the search input field is tapped into after opening the hamburger menu, a keyboard briefly appears and then disappears, and the menu closes.

![Mar-30-2021 11-27-33.gif](https://images.zenhubusercontent.com/5acae0cd4b5806bc2bce3338/ae1299eb-e58d-46a4-a1a4-a01cee0c5df1)

...and the following errors are thrown in the console:

![Screen Shot 2021-03-30 at 11.04.09 am.png](https://images.zenhubusercontent.com/5acae0cd4b5806bc2bce3338/21408e9f-9dd3-433c-af7f-3bd46de7922b)

#### MegaMenu.vue component

- The function `closeMobileIfDesktop()` appears to be firing (line 405)
- `closeMobileIfDesktop()` appears to be following its condition `if (this.isMobileOpen)` which contains code to dismiss and close the mobile hamburger menu
- we're not on desktop so it's not clear why this function is firing in the first place, so I've checked what might be firing this
- turns out there is an event listener which is listening for a browser resize on line 268 which is automatically closing the mobile menu on window resize, but only if the browser is perceived to be a "desktop" one (can't resize browsers on mobile generally speaking)
- further investigation reveals "desktop" is not defined anywhere but "mobile" is - line 257
- "desktop" therefore means anything which is not "mobile"
- "mobile" is being defined as ANYTHING with a width measurement of less than 900px
- the Samsung model in question has a raw resolution of 2960 x 1440 pixels (Height x Width)
- because mobiles generally have a very high raw resolution so pixels aren't visible, this "mobile" check is not functioning as intended
- it is unclear why the event listener is triggering a resize event in the first place - perhaps because the keyboard is appearing on screen?

#### Proposed fix

- the check for determining what is a "mobile" needs to be more robust than just checking the current width of the mega menu; many mobiles have higher resolution screens than desktops
- if it has to be based on resolution, it should be based on 'effective' resolution but I'm not sure if these values can easily be retrieved by a browser
